gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,163 @@
+        """Return a subset of images with valid exif.
+
+        Args:
+            search_path: image sequence search path.
+            max_num_imgs: the maximum number of images to process.
+
+        Returns:
+            Tuple[
+                List of selected image path.
+                The number of images with valid exif data.
+                The number of all the images.
+            ]
+        """
+        all_image_paths = glob.glob(search_path)
+        num_all_imgs = len(all_image_paths)
+        exif_image_paths = []
+        for single_img_path in all_image_paths:
+            # Drop images without valid exif data.
+            if io_utils.load_image(single_img_path).get_intrinsics_from_exif():
+                exif_image_paths.append(single_img_path)
+
+        # Randomly select a subset to process.
+        num_exif_imgs = len(exif_image_paths)
+        max_num_imgs = max_num_imgs if (max_num_imgs > 0 and max_num_imgs < num_exif_imgs) else num_exif_imgs
+        random_indices = np.random.permutation(num_exif_imgs)
+        return (
+            np.array(exif_image_paths)[random_indices[:max_num_imgs]],
+            num_exif_imgs,
+            num_all_imgs,
+        )
+
+    def image_filenames(self) -> List[str]:
+        """Return the file names corresponding to each image index."""
+        return [Path(fpath).name for fpath in self._image_paths]
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            The number of images.
+        """
+        return self._num_imgs
+
+    def get_image_full_res(self, index: int) -> Image:
+        """Get the image at the given index, at full resolution.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError("Image index is invalid")
+        return io_utils.load_image(self._image_paths[index])
+
+    def get_camera_intrinsics_full_res(self, index: int) -> Optional[Cal3Bundler]:
+        """Get the camera intrinsics at the given index, valid for a full-resolution image.
+
+        Args:
+            index: the index to fetch.
+
+        Returns:
+            Intrinsics for the given camera.
+        """
+        # Get intrinsics from exif.
+        intrinsics = io_utils.load_image(self._image_paths[index]).get_intrinsics_from_exif()
+        return intrinsics
+
+    def get_camera_pose(self, index: int) -> Optional[Pose3]:
+        """Get the camera pose (in world coordinates) at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Returns:
+            The camera pose w_T_index.
+        """
+        # No prior pose is available for images collected from Internet
```

Comment:
might say instead:

# The 1DSfM datasets do not provide ground truth or prior camera poses.

Reply:
Done.